lib/models/environment-variables.coffee

```diff
@@ -158,22 +159,23 @@ exports.device = {}
 # @public
 # @function
 #
-# @param {(String|Number)} deviceId - device id
+# @param {String} deviceName - device name
 # @param {module:resin.models.environment-variables.device~getAllCallback} callback - callback
 #
 # @example
-#	resin.models.environmentVariables.device.getAll 51, (error, environmentVariables) ->
+#	resin.models.environmentVariables.device.getAll 'MyDevice', (error, environmentVariables) ->
 #		throw error if error?
 #		console.log(environmentVariables)
 ###
-exports.device.getAll = (deviceId, callback) ->
-	return pine.get
-		resource: 'device_environment_variable'
-		options:
-			filter:
-				device: deviceId
-			expand: 'device'
-			orderby: 'env_var_name asc'
+exports.device.getAll = (deviceName, callback) ->
+	deviceModel.get(deviceName).then (device) ->
```

Comment:
I'm fairly sure if someone wanted to use the SDK to query the status of 10,000 devices that they'd appreciate being able to specify just the fields they're interested in.  Also it would be an optional thing so the default interface would remain the same, as for implementing it could be done using shared code to minimise the overhead, eg:
```
defaults = (options, callback) ->
	if _.isFunction(options)
		return { callback: options, options: {} }
	return { callback, options }

optionedGet = (options, query) ->
	if options?.select?
		if query.options?.select?
			throw new Error('Cannot specify select for this command')
		query.options ?= {}
		query.options.select = options.select
	return pine.get(query)

exports.device.getAll = (deviceName, options, callback) ->
	{ options, callback } = defaults(options, callback)
	deviceModel.get(deviceName).then (device) ->
		optionedGet options,
			resource: 'device_environment_variable'
			options:
				filter:
					device: device.id
				expand: 'device'
				orderby: 'env_var_name asc'
```